src/FieldHelper/RichTextFieldHelper.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+namespace Elbformat\FieldHelperBundle\FieldHelper;
+
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\ContentStruct;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\Core\Helper\FieldHelper;
+
+/**
+ * Helps reading, updating and comparing richtext field types.
+ * https://doc.ezplatform.com/en/latest/api/field_type_reference.
+ *
+ * @author Ronny Gericke <ronny.gericke@elbformat.de>
+ */
+class RichTextFieldHelper extends AbstractFieldHelper
+{
+    protected FieldHelper $fieldHelper;
+
+    public function __construct(FieldHelper $fieldHelper)
+    {
+        $this->fieldHelper = $fieldHelper;
+    }
+
+    public static function getName(): string
+    {
+        return self::class;
+    }
+
+    public function getXml(Content $content, string $fieldName): ?\DOMDocument
+    {
+        $field = $this->getField($content, $fieldName);
+
+        return $this->getXmlFieldValue($field);
+    }
+
+    public function isEmpty(Content $content, string $fieldName): bool
+    {
+        return $this->fieldHelper->isFieldEmpty($content, $fieldName);
+    }
+
+    public function updateXml(ContentStruct $struct, string $fieldName, ?\DOMDocument $value, ?Content $content): bool
+    {
+        // No changes
+        if (null !== $content) {
+            $field = $this->getField($content, $fieldName);
+            if ($this->isXmlEqual($field, $value)) {
+                return false;
+            }
+        }
+        $struct->setField($fieldName, $value);
+
+        return true;
+    }
+
+    protected function isXmlEqual(Field $field, ?\DOMDocument $value): bool
+    {
+        $fieldVal = $this->getXmlFieldValue($field);
+
+        return $fieldVal === $value;
```

Comment:
We want to check if objects are equal (containing the same content) and not if they are the same. I'm not quite sure if this works out-of-the-box (like for DateTime). It's safer to compare the `->saveXml()` values.